pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -244,26 +247,28 @@ func (s *DiscoveryServer) updateClusterInc(push *model.PushContext, clusterName
 	// for this cluster
 	for _, endpoints := range se.Shards {
 		for _, ep := range endpoints {
-			if svcPort.Name != ep.ServicePortName {
-				continue
-			}
-			// Port labels
-			if !labels.HasSubsetOf(model.Labels(ep.Labels)) {
-				continue
-			}
-			cnt++
+			for _, port := range ports {
```

Comment:
I donot think this is the root cause. As it seems similar to the previous. 

Previously even if `ports.GetByPort(clusterPort)` returns the udp port, it doesn't matter, because both the tcp and udp port are in `se.Shards`. So the port 53 must be in `LocalityLbEndpoints`.

So I think we haven't find the root cause yet.

Reply:
Udp port name doesn't match ServicePortName so it is skipped

Reply:
`for _, ep := range endpoints {`  in the endpoints, there is also a udp port endpoint, it can be matched, and it doesn't matter, as long as 53 is matched, it will exist in `LbEndpoints `.

Reply:
I think maybe endpoints doesn't have udp port endpoint because we filtering them out - not sure. Either way I have tested this and without this change it doesn't exist in LbEndpoints.

We should probably add some tests for this as well though

Reply:
I add some debug info, the udp port exist in lbEndpoints. We have no filter logic currently.

Reply:
I just double checked /endpointShardz from Karl's cluster, only the TCP port is there